mod-source-record-storage-server/src/main/java/org/folio/services/RecordServiceImpl.java

```diff
@@ -69,11 +73,27 @@
   }
 
   @Override
-  public Future<Boolean> saveRecords(RecordCollection recordCollection, String tenantId) {
-    ArrayList<Future> saveFutures = new ArrayList<>();
-    recordCollection.getRecords()
-      .forEach(record -> saveFutures.add(saveRecord(record, tenantId)));
-    return CompositeFuture.all(saveFutures).map(Future::succeeded);
+  public Future<RecordCollection> saveRecords(RecordCollection recordCollection, String tenantId) {
+    Map<Record, Future<Boolean>> savedRecords = recordCollection.getRecords().stream()
+      .map(record -> Pair.of(record, saveRecord(record, tenantId)))
+      .collect(LinkedHashMap::new, (map, pair) -> map.put(pair.getKey(), pair.getValue()), Map::putAll);
+
+    Future<RecordCollection> result = Future.future();
+
+    CompositeFuture.join(new ArrayList<>(savedRecords.values())).setHandler(ar -> {
+        RecordCollection records = new RecordCollection();
+        savedRecords.forEach((record, future) -> {
+          if (future.failed()) {
+            records.getErrorMessages().add(future.cause().getMessage());
```

Comment:
consider adding some logging here

Reply:
Good point! I decided to add logging in rest layer in order to avoid flooding in a log